### PR TITLE
0.17: Backport for #15085

### LINKED ISSUE #15079: gui: Incorrect application name when passing -regtest on Fedora
On Fedora 29.1-2, when running `bitcoin-qt -regtest`, the name of the application in the menu bar is shown as `Bitcoin-Qt-testnet` (the icon is the correct blue colour). i.e:
fedora - notifications

Noticed while testing #15000. When running just `bitcoin-qt`, or passing `-testnet` the menu name and icon are both correct.

### PR DESCRIPTION
**Pull request by benthecarman**
Changes the application name to `Bitcoin-Qt-regtest` when instead of `Bitcoin-Qt-testnet`

Fixes #15079

### CODE REVIEW

**Comment by fanquake** [comment]
You'll need to wait for #15085 to be merged first. Then you can create a proper backport commit, (see [this script](https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/master/backport.py) for help) which has the correct `Github-Pull`, `Rebased-From` info in the commit body. See #15002 for an example.

**Comment by benthecarman** [comment]
Oh okay, thank you

**Comment by promag** [comment]
> You'll need to wait for #15085 to be merged first

@fanquake didn't wait in #15002 :trollface:

**Comment by fanquake** [comment]
@benthecarman Could you now turn this into a proper backporting commit?

**Comment by promag** [comment]
@benthecarman still not right.

**Comment by MarcoFalke** [comment]
There is a script to do that: https://github.com/bitcoin-core/bitcoin-maintainer-tools#backport

**Review by fanquake — APPROVED** [review]
utACK 59dd855

**Comment by laanwj** [comment]
I don't think this warrants a separate backport PR. I think it's better to group these small changes, and leave separate backport PR for difficult changes that are significantly different between branches.

That said utACK 59dd855d2de77cd72a2b3190a825783ff3627e17 for this time.

**Pull request merged**
